cmd/minishift/cmd/root.go

```diff
@@ -46,7 +48,7 @@ import (
 )
 
 var dirs = [...]string{
```

Comment:
> Regarding initializing Viper and then checking profile does not seem to work for me. We can discuss about this.

what I am asking is why you initialize the variables at all. In the form you are using it, the _dirs_ will be initialized before any method executes. In our case this makes no sense anymore. We need to defer this until we know the profile. So in the case you want to keep the _dirs_ var it would be for example, _var dirs = []string{}_. You initialize it as empty slice/array. Then again, we are probably better of with a struct here. 

Note, you will need do be careful here as well in _initConfig_:

```go
    func initConfig() {
    	configPath := constants.ConfigFile
    	viper.SetConfigFile(configPath)
    	viper.SetConfigType("json")
    	err := viper.ReadInConfig()
    	if err != nil {
    		glog.Warningf("Error reading config file at %s: %s",     configPath, err)
    	}
    	setupViper()
    }
```

You need to determine wether the profile parameter is passed or which profile is selected in all instance config. Only once you know this, you can call to _viper.SetConfigFile(configPath)_